File: wear/src/main/java/com/gmits/gettingstarted/MainActivity.java
package com.gmits.gettingstarted;

import android.graphics.Color;
import android.os.Bundle;
import android.support.wear.widget.WearableLinearLayoutManager;
import android.support.wear.widget.WearableRecyclerView;
import android.support.wearable.activity.WearableActivity;
import android.support.wearable.view.WatchViewStub;

public class MainActivity extends WearableActivity {

    private WearableRecyclerView recycler_launcher_view;
    private CustomRecyclerAdapter customRecyclerAdapter;
    private CustomScrollingLayoutCallback customScrollingLayoutCallback;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        setAmbientEnabled();

        final WatchViewStub stub = findViewById(R.id.watch_view_stub);
        stub.setOnLayoutInflatedListener(new WatchViewStub.OnLayoutInflatedListener() {
            @Override
            public void onLayoutInflated(WatchViewStub stub) {

                recycler_launcher_view = stub.findViewById(R.id.recycler_launcher_view);
                recycler_launcher_view.setEdgeItemsCenteringEnabled(true);
                customScrollingLayoutCallback = new CustomScrollingLayoutCallback();
                WearableLinearLayoutManager wearableLinearLayoutManager = new WearableLinearLayoutManager(MainActivity.this, customScrollingLayoutCallback);
                recycler_launcher_view.setLayoutManager(wearableLinearLayoutManager);
                recycler_launcher_view.setCircularScrollingGestureEnabled(true);
                recycler_launcher_view.setBezelFraction(0.5f);
                recycler_launcher_view.setScrollDegreesPerScreen(90);

                customRecyclerAdapter = new CustomRecyclerAdapter();
                customRecyclerAdapter.setOnItemSelectListener(new CustomRecyclerAdapter.OnItemClick() {
                    @Override
                    public void onItemColorSelect(String colorCode) {
                        recycler_launcher_view.setBackgroundColor(Color.parseColor(colorCode));
                    }
                });

                recycler_launcher_view.setAdapter(customRecyclerAdapter);
            }
        });
    }
}

File: wear/src/main/java/com/gmits/gettingstarted/MainActivity.kt
package com.gmits.gettingstarted

import android.graphics.Color
import android.os.Bundle
import android.support.wear.widget.WearableLinearLayoutManager
import android.support.wear.widget.WearableRecyclerView
import android.support.wearable.activity.WearableActivity
import android.support.wearable.view.WatchViewStub

class MainActivity : WearableActivity() {

    private var recycler_launcher_view: WearableRecyclerView? = null
    private var customRecyclerAdapter: CustomRecyclerAdapter? = null
    private var customScrollingLayoutCallback: CustomScrollingLayoutCallback? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setAmbientEnabled()

        val stub = findViewById<WatchViewStub>(R.id.watch_view_stub)
        stub.setOnLayoutInflatedListener { stub ->
            recycler_launcher_view = stub.findViewById(R.id.recycler_launcher_view)
            recycler_launcher_view!!.isEdgeItemsCenteringEnabled = true
            customScrollingLayoutCallback = CustomScrollingLayoutCallback()
            val wearableLinearLayoutManager = WearableLinearLayoutManager(this@MainActivity, customScrollingLayoutCallback)
            recycler_launcher_view!!.layoutManager = wearableLinearLayoutManager
            recycler_launcher_view!!.isCircularScrollingGestureEnabled = true
            recycler_launcher_view!!.bezelFraction = 0.5f
            recycler_launcher_view!!.scrollDegreesPerScreen = 90f

            customRecyclerAdapter = CustomRecyclerAdapter()
            customRecyclerAdapter!!.setOnItemSelectListener(object : CustomRecyclerAdapter.OnItemClick {
                override fun onItemColorSelect(colorCode: String) {
                    recycler_launcher_view!!.setBackgroundColor(Color.parseColor(colorCode))
                }
            })

            recycler_launcher_view!!.adapter = customRecyclerAdapter
        }
    }
}
